code/modules/mob/living/brain/brain_item.dm

```diff
@@ -146,3 +146,89 @@
 	desc = "We barely understand the brains of terrestial animals. Who knows what we may find in the brain of such an advanced species?"
 	icon_state = "brain-x"
 	origin_tech = "biotech=6"
+
+// IPC brain fuckery.
+/obj/item/organ/brain/mmi_holder
+	name = "brain"
+	slot = "brain"
+	zone = "chest"
+	status = ORGAN_ROBOTIC
+	remove_on_qdel = FALSE
+	var/obj/item/device/mmi/stored_mmi
+
+/obj/item/organ/brain/mmi_holder/Destroy()
+	QDEL_NULL(stored_mmi)
+	return ..()
+
+/obj/item/organ/brain/mmi_holder/Insert(mob/living/carbon/C, special = 0,no_id_transfer = FALSE)
+
+	owner = C
+	C.internal_organs |= src
+	C.internal_organs_slot[slot] = src
+	loc = null
+	//the above bits are copypaste from organ/proc/Insert, because I couldn't go through the parent here.
+
+	if(stored_mmi.brainmob)
+		if(C.key)
+			C.ghostize()
+		var/mob/living/brain/B = stored_mmi.brainmob
+		if(stored_mmi.brainmob.mind)
+			B.mind.transfer_to(C)
+		else
+			C.key = B.key
+
+	if(ishuman(C))
+		var/mob/living/carbon/human/H = C
+		if(H.dna && H.dna.species && (REVIVESBYHEALING in H.dna.species.species_traits))
+			if(H.health > 0)
+				H.revive(0)
+	//	QDEL_NULL(stored_mmi.brainmob)
+	// :v
+
+	update_from_mmi()
+
+/obj/item/organ/brain/mmi_holder/Remove(var/mob/living/user, special = 0)
+	if(!special)
+		if(stored_mmi)
+			. = stored_mmi
+			if(owner.mind)
+				owner.mind.transfer_to(stored_mmi.brainmob)
+			stored_mmi.loc = owner.loc
+			if(stored_mmi.brainmob)
+				var/mob/living/brain/B = stored_mmi.brainmob
+				spawn(0)
```

Comment:
Because there's fancy shmancy stuff going on in the surgery code. We want to make sure we toggle it after the mind transfer and holder deletion happen properly. It didn't work reliably without the spawn.